### PR TITLE
Feat: bridge module for `SynapseBridge`

### PR DESCRIPTION
**Pull request by ChiTimesChi**
## Description

Adds a Bridge Module for `SynapseBridge` contract to be used with new `SynapseRouterV2` contract. It is treating older `SynapseRouter` deployment as the local bridge config contract, and reimplements the bridging / getters logic from scratch.

Leaving links for reference:
- https://github.com/synapsecns/synapse-contracts/blob/master/contracts/bridge/router/SynapseRouter.sol
- https://github.com/synapsecns/synapse-contracts/blob/master/contracts/bridge/router/LocalBridgeConfig.sol

## Checklist

- [ ] New Contracts have been tested
- [ ] Lint has been run
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: contracts/router/modules/bridge/SynapseBridgeModule.sol [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+    }
+
+    /// @inheritdoc IBridgeModule
+    function delegateBridge(
+        address to,
+        uint256 chainId,
+        address token,
+        uint256 amount,
+        SwapQuery memory destQuery
+    ) external payable {
```
Comment (mikeyrf) [inline contracts/router/modules/bridge/SynapseBridgeModule.sol:35]: Any thoughts on removing `payable` given the delegate call. Unclear we need it and reminds me of this issue every time I see it https://github.com/Uniswap/v3-periphery/issues/52 (although not the same as multicall here)

File: contracts/router/modules/bridge/SynapseBridgeModule.sol [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+        assertDelegateCall();
+        (ILocalBridgeConfig.TokenType tokenType, address bridgeToken) = localBridgeConfig.config(token);
+        // Use config.bridgeToken as the token address for the bridging purposes
+        if (bridgeToken == address(0)) revert SynapseBridgeModule__UnsupportedToken(token);
+        // Approve the bridge to spend the token
+        bridgeToken.universalApproveInfinity({spender: address(synapseBridge), amountToSpend: amount});
+        if (destQuery.routerAdapter == address(0)) {
+            // If no Router Adapter is set, no action is required on destination chain, use `deposit()` or `redeem()`
+            (tokenType == ILocalBridgeConfig.TokenType.Redeem ? synapseBridge.redeem : synapseBridge.deposit)(
```
Comment (mikeyrf) [inline contracts/router/modules/bridge/SynapseBridgeModule.sol:44]: Even though this looks cool in choosing function sig bw `redeem` and `deposit`, might not be easiest to read

File: contracts/router/modules/bridge/SynapseBridgeModule.sol (hunk 1 above)
Comment (mikeyrf) [inline contracts/router/modules/bridge/SynapseBridgeModule.sol:35]: Any thoughts on removing `payable` given the delegate call? Unclear to me we need it for the bridge module and reminds me of this issue every time I see it https://github.com/Uniswap/v3-periphery/issues/52 (although not the same as multicall here)

File: contracts/router/modules/bridge/SynapseBridgeModule.sol [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+                to,
+                chainId,
+                bridgeToken,
+                amount
+            );
+        } else {
+            // Otherwise, use tokenType-dependent internal function to give destination chain action instructions
+            (tokenType == ILocalBridgeConfig.TokenType.Redeem ? _redeemTokenWithAction : _depositTokenWithAction)(
```
Comment (mikeyrf) [inline contracts/router/modules/bridge/SynapseBridgeModule.sol:52]: Same as above here with function choice

**Review by mikeyrf — APPROVED** [review]
Small comments but otherwise lgtm

File: contracts/router/modules/bridge/SynapseBridgeModule.sol (hunk 1 above)
Comment (ChiTimesChi) [reply contracts/router/modules/bridge/SynapseBridgeModule.sol:35]: We need the `payable` here, as the RouterV2 main entry function is payable (user can start from native ETH). In the context of delegate call module and Router are sharing `msg.value`, so removing `payable` leads to failed tests:
```
[FAIL. Reason: Expected an emit, but the call reverted instead. Ensure you're testing the happy path when using the `expectEmit` cheatcode.] testDelegateBridgeMsgValueDeposit() (gas: 454483)
Traces:
  [454483] SynapseBridgeModuleTest::testDelegateBridgeMsgValueDeposit()
    ├─ ...
    ├─ [5090] DelegateCaller::performDelegateCall{value: 2000000000000000000}(SynapseBridgeModule: [0x5991A2dF15A8F6A256D3Ec51E99254Cd3fb576A9], 0x436f3aa5000000000000000000000000000000000000000000000000000000000000000a0000000000000000000000000000000000000000000000000000000000000014000000000000000000000000c7183455a4c133ae270771860664b6b7ec320bb10000000000000000000000000000000000000000000000000de0b6b3a764000000000000000000000000000000000000000000000000000000000000000000a0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000a00000000000000000000000000000000000000000000000000000000000000000) 
    │   ├─ [45] SynapseBridgeModule::delegateBridge(0x000000000000000000000000000000000000000A, 20, DT: [0xc7183455a4C133Ae270771860664b6B7ec320bB1], 1000000000000000000 [1e18], (0x0000000000000000000000000000000000000000, 0x0000000000000000000000000000000000000000, 0, 0, 0x)) [delegatecall]
    │   │   └─ ← "EvmError: Revert"
    │   └─ ← "Address: low-level delegate call failed"
```

File: contracts/router/modules/bridge/SynapseBridgeModule.sol (hunk 2 above)
Comment (ChiTimesChi) [reply contracts/router/modules/bridge/SynapseBridgeModule.sol]: Resolved in 2883557e2ad03599c6c9757b3fd561283a62bed6

File: contracts/router/modules/bridge/SynapseBridgeModule.sol (hunk 1 above)
Comment (mikeyrf) [reply contracts/router/modules/bridge/SynapseBridgeModule.sol:36]: Understood. But thought at the router level user is supposed to use the adapter to deposit native ETH to WETH prior to interacting with bridge?

**Pull request merged**